### PR TITLE
NXP-24754: provide s3 direct upload config

### PR DESCRIPTION
**Pull request by pierre-gautier**

### CODE REVIEW

File: ear/src/main/resources/s3binaries/nuxeo.defaults [hunk 1]
```diff
@@ -12,3 +13,10 @@ nuxeo.s3storage.crypt.key.alias=
 nuxeo.s3storage.crypt.key.password=
 
 nuxeo.core.binarymanager=org.nuxeo.ecm.core.storage.sql.S3BinaryManager
+
+nuxeo.s3storage.useDirectUpload=true
```
Comment (covolution) [inline ear/src/main/resources/s3binaries/nuxeo.defaults]: Should it be true by default?

File: ear/src/main/resources/s3binaries/nxserver/config/s3directupload-config.xml.nxftl [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+<component name="s3DirectUpload" version="1.0.0">
+
+<#if "${nuxeo.s3storage.useDirectUpload}" == "true">
+
+  <require>org.nuxeo.ecm.core.storage.cloud.managment.contrib</require>
+
+  <extension target="org.nuxeo.ecm.core.transientstore.TransientStorageComponent" point="store">
+    <store name="s3TransientStore" class="org.nuxeo.ecm.core.transientstore.keyvalueblob.KeyValueBlobTransientStore">
+      <targetMaxSizeMB>1</targetMaxSizeMB>
```
Comment (efge) [inline ear/src/main/resources/s3binaries/nxserver/config/s3directupload-config.xml.nxftl]: This seems a bit small, and we should give admins an easy way to customize this. Please make this a property too.

File: ear/src/main/resources/s3binaries/nuxeo.defaults [hunk 3]
```diff
@@ -12,3 +13,10 @@ nuxeo.s3storage.crypt.key.alias=
...
+
+nuxeo.s3storage.direct.bucket=
```
Comment (efge) [inline ear/src/main/resources/s3binaries/nuxeo.defaults]: "direct" is not very meaningful on its own. I'd like a name that's about the role of the bucket itself in the system, not the software component that's using it. Maybe `nuxeo.s3storage.transient.bucket`?

File: ear/src/main/resources/s3binaries/nxserver/config/s3directupload-config.xml.nxftl [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+  <extension target="org.nuxeo.ecm.core.blob.BlobManager" point="configuration">
+    <blobprovider name="s3TransientStore">
+      <class>org.nuxeo.ecm.core.storage.sql.S3BinaryManager</class>
+      <property name="awsid">${nuxeo.s3storage.direct.awsid}</property>
+      <property name="awssecret">${nuxeo.s3storage.direct.awssecret}</property>
+      <property name="bucket">${nuxeo.s3storage.direct.bucket}</property>
+      <property name="region">${nuxeo.s3storage.direct.region}</property>
+      <property name="bucket.prefix">/</property>
+      <property name="transient">true</property>
```
Comment (efge) [inline ear/src/main/resources/s3binaries/nxserver/config/s3directupload-config.xml.nxftl]: This one is important and not related to S3. Let's separate it from the others and add a comment like `<!-- this blob provider is used for transient storage -->`

File: ear/src/main/resources/s3binaries/nxserver/config/s3directupload-config.xml.nxftl [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+  </extension>
+
+  <extension target="org.nuxeo.ecm.core.blob.BlobManager" point="configuration">
+    <blobprovider name="s3TransientStore">
+      <class>org.nuxeo.ecm.core.storage.sql.S3BinaryManager</class>
+      <property name="awsid">${nuxeo.s3storage.direct.awsid}</property>
+      <property name="awssecret">${nuxeo.s3storage.direct.awssecret}</property>
+      <property name="bucket">${nuxeo.s3storage.direct.bucket}</property>
+      <property name="region">${nuxeo.s3storage.direct.region}</property>
+      <property name="bucket.prefix">/</property>
```
Comment (efge) [inline ear/src/main/resources/s3binaries/nxserver/config/s3directupload-config.xml.nxftl]: The default prefix should be empty, not `/`.

File: ear/src/main/resources/s3binaries/nxserver/config/s3directupload-config.xml.nxftl [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      <property name="transient">true</property>
+      <property name="cachesize">100MB</property>
+      <property name="connection.max">50</property>
+      <property name="connection.retry">3</property>
+      <property name="directdownload">true</property>
+      <property name="directdownload.expire">3600</property>
```
Comment (efge) [inline ear/src/main/resources/s3binaries/nxserver/config/s3directupload-config.xml.nxftl]: Are we using direct download from that bucket? If not, let's not add the option.

**Review by efge — CHANGES_REQUESTED** [review]

File: ear/src/main/resources/s3binaries/nuxeo.defaults (hunk 1 above)
Comment (pierre-gautier) [reply ear/src/main/resources/s3binaries/nuxeo.defaults]: I believe it should, since it has advantages without drawbacks (I can see, at least)

Comment (efge) [reply ear/src/main/resources/s3binaries/nuxeo.defaults]: It requires additional config, so let's set it `false` by default.

File: ear/src/main/resources/s3binaries/nxserver/config/s3directupload-config.xml.nxftl [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+      <property name="awssecret">${nuxeo.s3storage.transient.awssecret}</property>
+      <property name="bucket">${nuxeo.s3storage.transient.bucket}</property>
+      <property name="region">${nuxeo.s3storage.transient.region}</property>
+      <property name="bucket.prefix"></property>
+      <property name="cachesize">100MB</property>
+      <property name="connection.max">50</property>
+      <property name="connection.retry">3</property>
+      <property name="connection.timeout">50000</property>
+      <property name="socket.timeout">50000</property>
+	  <!-- this blob provider is used for transient storage -->
```
Comment (efge) [inline ear/src/main/resources/s3binaries/nxserver/config/s3directupload-config.xml.nxftl]: Could you remove the tab here

**Review by efge — APPROVED** [review]

**Comment by efge** [comment]
👍

**Pull request merged**